debugger/src/server/session_data.rs

```diff
@@ -25,7 +25,7 @@ pub enum BreakpointType {
     InstructionBreakpoint,
     /// A breakpoint was requested using a source location, and usually a result of a user requesting a
     /// breakpoint while in a 'source' view.
-    SourceBreakpoint(Source, SourceLocation),
+    SourceBreakpoint(Source, Option<SourceLocation>),
```

Comment:
Can you describe what `None` means? Also, I would maybe make this named arguments rather than a tuple :)

Reply:
I've updated the doc comment.

Also, this is used only in 6 places in this crate, so instead of changing from the tuple, I've elected to make this enum private to this crate. It is convenient to have the enum tuple containing that data for when `ActiveBreakpoint` is cached.

Reply:
I am not sure I follow. I don't really like usage of Options and Tuples that are obfuscated.

I would make this a

```
enum BreakpointType {
...
    SourceBreakpoint {
        source: Source,
        location: SourceLocation
}
enum SourceLocation {
    All,
    Specific(SpecificSourceLocation)
}
```

then it's all way more obvious :)